lib/Login.dart

```diff
@@ -1,120 +1,52 @@
-import 'dart:convert';
-import 'package:carriage/app_config.dart';
 import 'package:flutter/material.dart';
-import 'Home.dart';
-import 'package:google_sign_in/google_sign_in.dart';
-//import 'app_config.dart';
-import 'main_common.dart';
+import 'package:provider/provider.dart';
+import 'AuthProvider.dart';
 
-GoogleSignIn googleSignIn = GoogleSignIn(
-  scopes: [
-    'email',
-    'https://www.googleapis.com/auth/userinfo.profile',
-  ],
-);
 
-_handleSignIn() async {
-  try {
-    await googleSignIn.signIn();
-  } catch (error) {
-    print(error);
-  }
-}
-
-Future<String> tokenFromAccount(GoogleSignInAccount account) async {
-  GoogleSignInAuthentication auth;
-  try {
-    auth = await account.authentication;
-    print('okay');
-  } catch (error) {
-    return null;
-  }
-  return auth.idToken;
-}
 
 class Login extends StatefulWidget {
```

Comment:
It might be unnecessary for Login to be Stateful now since it has nothing to keep track of on its own. Try converting this to a Stateless widget and seeing if functionality is preserved.